### PR TITLE
Add Yunet detector

### PR DESCRIPTION
**Pull request by AnthraX1**
Yunet is an ultra high performance face detector. For more details read articles below:
https://medium.com/@silkworm/yunet-ultra-high-performance-face-detection-in-opencv-a-good-solution-for-real-time-poc-b01063e251d5
https://github.com/ShiqiYu/libfacedetection.train/tree/master

I also added the ability to set the score_threshold(confidence threshold, default to 0.9) via setting env var "yunet_score_threshold".

### CODE REVIEW

**Comment by serengil** [comment]
Would you please add this detector in read me?

File: deepface/detectors/YunetWrapper.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    score_threshold = os.environ.get("yunet_score_threshold", score_threshold)
+    resp = []
+    detected_face = None
+    img_region = [0, 0, image.shape[1], image.shape[0]]
+    faces = []
+    height, width = image.shape[0], image.shape[1]
+    # resize image if it is too large (Yunet fails to detect faces on large input sometimes)
+    # I picked 640 as a threshold because it is the default value of max_size in Yunet.
+    if height > 640 or width > 640:
+        r = 640.0 / max(height, width)
```
Comment (serengil) [inline deepface/detectors/YunetWrapper.py:34]: If you resize the image, you need to restore it after detection. Coordinates should be updated, too.

File: deepface/detectors/YunetWrapper.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    _, faces = detector.detect(image)
+    if faces is None:
+        return resp
+    for face in faces:
+        """
+        The detection output faces is a two-dimension array of type CV_32F,
+        whose rows are the detected face instances, columns are the location of a face and 5 facial landmarks.
+        The format of each row is as follows:
+        x1, y1, w, h, x_re, y_re, x_le, y_le, x_nt, y_nt, x_rcm, y_rcm, x_lcm, y_lcm,
+        where x1, y1, w, h are the top-left coordinates, width and height of the face bounding box,
```
Comment (serengil) [inline deepface/detectors/YunetWrapper.py:48]: Please use meaningful variable names. Instead re, you can use right_eye.

Comment (AnthraX1) [reply deepface/detectors/YunetWrapper.py:48]: That's copied from opencv docs here:
https://docs.opencv.org/4.x/df/d20/classcv_1_1FaceDetectorYN.html

I think it should be in line with the vars in underlying cpp lib.

Comment (serengil) [reply deepface/detectors/YunetWrapper.py:48]: Okay. Mine is just recommendation. Feel free to discard this.

File: deepface/detectors/YunetWrapper.py (hunk 1 above)
Comment (AnthraX1) [reply deepface/detectors/YunetWrapper.py:35]: Fixed in latest commit. Tested on some large images and appeared work. Please check.

**Comment by AnthraX1** [comment]
> Would you please add this detector in read me?

Updated. You probably need to change the images too.

**Comment by serengil** [comment]
> > Would you please add this detector in read me?
> 
> 
> 
> Updated. You probably need to change the images too.

What do you mean? What should i do?

**Comment by serengil** [comment]
Oh you mean stock photos. Okay I will do it.

**Comment by serengil** [comment]
Thank you for your contribution.

**Comment by serengil** [comment]
It fails in the linting step:

2023-07-08T15:44:32.4937592Z pylint: Command line or configuration file:1: UserWarning: Specifying exception names in the overgeneral-exceptions option without module name is deprecated and support for it will be removed in pylint 3.0. Use fully qualified name (maybe 'builtins.BaseException' ?) instead.
2023-07-08T15:44:32.4940375Z pylint: Command line or configuration file:1: UserWarning: Specifying exception names in the overgeneral-exceptions option without module name is deprecated and support for it will be removed in pylint 3.0. Use fully qualified name (maybe 'builtins.Exception' ?) instead.
2023-07-08T15:44:32.4941882Z ************* Module deepface.detectors.SsdWrapper
2023-07-08T15:44:32.4943902Z deepface/detectors/SsdWrapper.py:37:20: I1101: Module 'cv2.dnn' has no 'readNetFromCaffe' member, but source is unavailable. Consider adding this module to extension-pkg-allow-list if you want to perform analysis based on run-time introspection of living objects. (c-extension-no-member)
2023-07-08T15:44:32.4946614Z deepface/detectors/SsdWrapper.py:71:16: I1101: Module 'cv2.dnn' has no 'blobFromImage' member, but source is unavailable. Consider adding this module to extension-pkg-allow-list if you want to perform analysis based on run-time introspection of living objects. (c-extension-no-member)
2023-07-08T15:44:32.4948079Z ************* Module deepface.detectors.YunetWrapper
2023-07-08T15:44:32.4949124Z deepface/detectors/YunetWrapper.py:9:0: C0301: Line too long (119/100) (line-too-long)
2023-07-08T15:44:32.4950234Z deepface/detectors/YunetWrapper.py:48:0: C0301: Line too long (110/100) (line-too-long)
2023-07-08T15:44:32.4951596Z deepface/detectors/YunetWrapper.py:52:0: C0301: Line too long (158/100) (line-too-long)
2023-07-08T15:44:32.4952838Z deepface/detectors/YunetWrapper.py:46:8: W0105: String statement has no effect (pointless-string-statement)
2023-07-08T15:44:32.4954289Z deepface/detectors/YunetWrapper.py:65:21: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
2023-07-08T15:44:32.4955845Z deepface/detectors/YunetWrapper.py:2:0: C0411: standard import "import os" should be placed before "import cv2" (wrong-import-order)
2023-07-08T15:44:32.4956488Z 
2023-07-08T15:44:32.4956825Z -----------------------------------
2023-07-08T15:44:32.4957380Z Your code has been rated at 9.97/10
2023-07-08T15:44:32.4957710Z 
2023-07-08T15:44:36.4364966Z ##[error]Process completed with exit code 20.

**Comment by AnthraX1** [comment]
> It fails in the linting step:
> 
> 2023-07-08T15:44:32.4937592Z pylint: Command line or configuration file:1: UserWarning: Specifying exception names in the overgeneral-exceptions option without module name is deprecated and support for it will be removed in pylint 3.0. Use fully qualified name (maybe 'builtins.BaseException' ?) instead. 2023-07-08T15:44:32.4940375Z pylint: Command line or configuration file:1: UserWarning: Specifying exception names in the overgeneral-exceptions option without module name is deprecated and support for it will be removed in pylint 3.0. Use fully qualified name (maybe 'builtins.Exception' ?) instead. 2023-07-08T15:44:32.4941882Z ************* Module deepface.detectors.SsdWrapper 2023-07-08T15:44:32.4943902Z deepface/detectors/SsdWrapper.py:37:20: I1101: Module 'cv2.dnn' has no 'readNetFromCaffe' member, but source is unavailable. Consider adding this module to extension-pkg-allow-list if you want to perform analysis based on run-time introspection of living objects. (c-extension-no-member) 2023-07-08T15:44:32.4946614Z deepface/detectors/SsdWrapper.py:71:16: I1101: Module 'cv2.dnn' has no 'blobFromImage' member, but source is unavailable. Consider adding this module to extension-pkg-allow-list if you want to perform analysis based on run-time introspection of living objects. (c-extension-no-member) 2023-07-08T15:44:32.4948079Z ************* Module deepface.detectors.YunetWrapper 2023-07-08T15:44:32.4949124Z deepface/detectors/YunetWrapper.py:9:0: C0301: Line too long (119/100) (line-too-long) 2023-07-08T15:44:32.4950234Z deepface/detectors/YunetWrapper.py:48:0: C0301: Line too long (110/100) (line-too-long) 2023-07-08T15:44:32.4951596Z deepface/detectors/YunetWrapper.py:52:0: C0301: Line too long (158/100) (line-too-long) 2023-07-08T15:44:32.4952838Z deepface/detectors/YunetWrapper.py:46:8: W0105: String statement has no effect (pointless-string-statement) 2023-07-08T15:44:32.4954289Z deepface/detectors/YunetWrapper.py:65:21: C0209: Formatting a regular string which could be a f-string (consider-using-f-string) 2023-07-08T15:44:32.4955845Z deepface/detectors/YunetWrapper.py:2:0: C0411: standard import "import os" should be placed before "import cv2" (wrong-import-order) 2023-07-08T15:44:32.4956488Z 2023-07-08T15:44:32.4956825Z ----------------------------------- 2023-07-08T15:44:32.4957380Z Your code has been rated at 9.97/10 2023-07-08T15:44:32.4957710Z 2023-07-08T15:44:36.4364966Z ##[error]Process completed with exit code 20.

I hate overzealous linters :)

**Comment by serengil** [comment]
Never mind. I will it. It is not blocking issue.

I should run actions before merging :)

**Pull request merged**